src/datasources/test-plans/types.ts

```diff
@@ -14,61 +14,166 @@ export enum OutputType {
 }
 
 export enum Properties {
-    assignedTo = 'Assigned to',
-    createdAt = 'Created at',
-    createdBy = 'Created by',
-    description = 'Description',
-    dueDate = 'Due date',
-    earliestStartDate = 'Earliest start date',
-    id = 'ID',
-    name = 'Name',
-    properties = 'Properties',
-    requestedBy = 'Requested by',
-    state = 'State',
-    type = 'Type',
-    updatedAt = 'Updated at',
-    updatedBy = 'Updated by',
-    workspace = 'Workspace',
-    workOrder = 'Work order name (Work oder ID)',
-    product = 'Product (Part number)',
-    plannedStartDate = 'Planned start date',
-    estimatedEndDate = 'Estimated end date',
-    estimatedDuration = 'Estimated duration',
-    system = 'System',
-    template = 'Test plan template (Template ID)',
-    testProgram = 'Test program name',
-    substate = 'Substate',
-    fixtureIds = 'Fixture IDs',
-    dut = 'DUT',
-}
+    ASSIGNED_TO = 'ASSIGNED_TO',
+    CREATED_AT = 'CREATED_AT',
+    CREATED_BY = 'CREATED_BY',
+    DESCRIPTION = 'DESCRIPTION',
+    ID = 'ID',
+    NAME = 'NAME',
+    PROPERTIES = 'PROPERTIES',
+    STATE = 'STATE',
+    UPDATED_AT = 'UPDATED_AT',
+    UPDATED_BY = 'UPDATED_BY',
+    WORKSPACE = 'WORKSPACE',
+    WORK_ORDER = 'WORK_ORDER',
+    WORK_ORDER_ID = 'WORK_ORDER_ID',
+    PRODUCT = 'PRODUCT',
+    PLANNED_START_DATE_TIME = 'PLANNED_START_DATE_TIME',
+    ESTIMATED_END_DATE_TIME = 'ESTIMATED_END_DATE_TIME',
+    ESTIMATED_DURATION_IN_SECONDS = 'ESTIMATED_DURATION_IN_SECONDS',
+    SYSTEM_NAME = 'SYSTEM_NAME',
+    SYSTEM_ID = 'SYSTEM_ID',
+    TEMPLATE = 'TEMPLATE',
+    TEMPLATE_ID = 'TEMPLATE_ID',
+    TEST_PROGRAM = 'TEST_PROGRAM',
+    SUBSTATE = 'SUBSTATE',
+    FIXTURE_NAMES = 'FIXTURE_NAMES',
+    DUT_ID = 'DUT_ID',
+};
 
-export const PropertiesOptions = {
-    ASSIGNED_TO: 'assignedTo',
-    CREATED_AT: 'createdAt',
-    CREATED_BY: 'createdBy',
-    DESCRIPTION: 'description',
-    DUE_DATE: 'dueDate',
-    EARLIEST_START_DATE: 'earliestStartDate',
-    ID: 'id',
-    NAME: 'name',
-    PROPERTIES: 'properties',
-    REQUESTED_BY: 'requestedBy',
-    STATE: 'state',
-    TYPE: 'type',
-    UPDATED_AT: 'updatedAt',
-    UPDATED_BY: 'updatedBy',
-    WORKSPACE: 'workspace',
-    WORK_ORDER: 'workOrder',
-    PRODUCT: 'product',
-    PLANNED_START_DATE: 'plannedStartDate',
-    ESTIMATED_END_DATE: 'estimatedEndDate',
-    ESTIMATED_DURATION: 'estimatedDuration',
-    SYSTEM: 'system',
-    TEMPLATE: 'template',
-    TEST_PROGRAM: 'testProgram',
-    SUBSTATE: 'substate',
-    FIXTURE_IDS: 'fixtureIds',
-    DUT: 'dut',
+export enum TimeFields {
+    CREATED_AT = 'CREATED_AT',
+    UPDATED_AT = 'UPDATED_AT',
+    PLANNED_START_DATE_TIME = 'PLANNED_START_DATE_TIME',
+    ESTIMATED_END_DATE_TIME = 'ESTIMATED_END_DATE_TIME'
+};
+
+export const PropertiesProjectionMap: Record<Properties, { label: string, projection: string[], field: string[] }> = {
```

Comment:
We should improve the typing of this object instead of just being "string".

I'm making a guess, but looks like the "projection" is associated with the Properties enum and the "field" is TestPlanResponseProperties. If that is the case we can get typescript to represent that as:

```suggestion
export const PropertiesProjectionMap: Record<Properties, {
    label: string,
    projection: (keyof typeof Properties)[],
    field: (keyof TestPlanResponseProperties)[]
}> = {
```

And if that is accurate then it looks like the object has a few errors:

![image](https://github.com/user-attachments/assets/218aa4de-f910-4d12-85f6-67a710ed35e9)

![image](https://github.com/user-attachments/assets/fff59d56-7a4d-457d-8d25-d143613c91ec)

Reply:
The assumption is partially correct that `projection` is not `(keyof typeof Properties)`, because the type `Properties` used to show UI dropdown, but the actual API does not allow those values as projection. I will create a separate enum for `Projection` and use it here.